Add introduction banner card into multichannel marketing page

ecgan opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34904. Previous draft PR is PR https://github.com/woocommerce/woocommerce/pull/36220.

In this PR, we add the introduction banner card at the top of the multichannel marketing page.

There are two "types" of introduction banner card: one with the "Create a campaign" and "Add channels" buttons, and one without.

Screenshot of introduction banner card without the "Create a campaign" and "Add channels" buttons:

![image](https://user-images.githubusercontent.com/417342/209806777-a220e7f2-8937-45b2-bc07-726a2d0a2b19.png)

Screenshot of introduction banner card with the "Create a campaign" and "Add channels" buttons:

![image](https://user-images.githubusercontent.com/417342/209806644-dad37194-5224-4f60-98b0-353c9d1eb7a8.png)

Users can dismiss the card by clicking at the top right "X" button. We store the dismissed state by using a WP Option `woocommerce_marketing_overview_multichannel_banner_dismissed`. This approach is similar to the existing [`woocommerce_marketing_overview_welcome_hidden`](https://github.com/woocommerce/woocommerce/blob/3daeac9a698f71f84619a37335d5289598a5b394/plugins/woocommerce-admin/client/marketing/overview/welcome-card/index.js#L21-L23) WP Option for the WelcomeCard in the existing Marketing page.

The "Creata a campaign" button is displayed when there is at least one registered channel. Clicking it will display the "Create a campaign" modal.

The "Add channels" button is displayed when there is at least one registered channel and one recommended channel. When users click on the "Add channels" button, the page will scroll down to the "Add channels" button in the Channels card, allowing users to install a new channel.

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

Pre-conditions: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
- If you have dismissed the banner card, you can make it show up again by deleting the WP Option `woocommerce_marketing_overview_multichannel_banner_dismissed` in the database.

1. Go to WooCommerce Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. You should see the introduction banner card at the top of the page.
3. Proceed to setup a channel, and then come back to this marketing page.
4. Click on the "Create campaign" button. You should see the "Create a campaign" modal.
5. Click on the "Add channels" button. The page should scroll down to the "Add channels" button in the Channels card.
6. Dismiss the card, and reload the page. The card should not be displayed.

<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

ecgan committed:
Add introduction banner card to marketing page.

ecgan committed:
Add close button to introduction banner card.

ecgan committed:
Display buttons in introduction banner with large illustration.

ecgan committed:
Display and dismiss introduction banner with wp_options.

ecgan committed:
Rename UseIntroductionBannerType to UseIntroductionBanner.

ecgan committed:
Focus on Add Channels section when users click on Add Channels button.

ecgan committed:
Code refactor with useIsLocationHashAddChannels.

ecgan committed:
Merge branch 'feature/34909-marketing-create-campaign-modal' into feature/34904-marketing-introduction-banner

Conflicts:
	plugins/woocommerce-admin/client/marketing/hooks/index.ts
	plugins/woocommerce-admin/client/marketing/overview-multichannel/Channels/RecommendedChannels.tsx
	plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx

ecgan committed:
Move CreateNewCampaignModal to shared components folder.

ecgan committed:
Display CreateNewCampaignModal upon button click in IntroductionBanner.

ecgan committed:
Use ref instead of location hash to scroll to add channels section.

ecgan commented:
@puntope , this PR is build on top of PR https://github.com/woocommerce/woocommerce/pull/37044, could you also help to review this PR please? 🙏

ecgan committed:
Better logic for displaying buttons in IntroductionBanner.

ecgan committed:
Merge branch 'feature/34909-marketing-create-campaign-modal' into feature/34904-marketing-introduction-banner

ecgan committed:
Add changelog.

ecgan committed:
Run lint:css-fix.

ecgan committed:
Fix failed test for Campaigns component.

eason9487 commented on plugins/woocommerce-admin/client/marketing/hooks/useIntroductionBanner.ts:
```diff
@@ -0,0 +1,40 @@
+		const isUpdateRequesting = isOptionsUpdating();
+
+		return {
+			loading: ! hasFinishedResolution( 'getOption', [ OPTION_NAME ] ),
```
Precisely, `! hasFinishedResolution` is not exactly equivalent to the loading status, although it works correctly here but is not very precise in the semantic meaning.

ecgan replied:
Related discussions in https://github.com/woocommerce/woocommerce/pull/37044#discussion_r1139197088. I can come up with a separate PR later to change the naming on this to `hasResolved`, `hasFinished` or something else.

eason9487 commented on plugins/woocommerce-admin/client/marketing/hooks/useIntroductionBanner.ts:
```diff
@@ -0,0 +1,40 @@
+				getOption( OPTION_NAME ) === OPTION_VALUE || isUpdateRequesting,
+			dismissIntroductionBanner,
+		};
+	} );
```
It would be a bit nicer to specify `deps` for `useSelect` in order to memorize the return.

```suggestion
	}, [] );
```

ecgan replied:
Thanks. I committed your suggestion in [`4925342` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/49253425b213c97b2a855854055a249817b869dc). Then, I found there is actually a dependency on `dismissIntroductionBanner`, so I made another change in [`1b65be0` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/1b65be0a5338f3b235f048fa7da29a42330e3227).

eason9487 commented on plugins/woocommerce-admin/client/marketing/hooks/useIntroductionBanner.ts:
```diff
@@ -0,0 +1,40 @@
+
+const OPTION_NAME =
+	'woocommerce_marketing_overview_multichannel_banner_dismissed';
+const OPTION_VALUE = 'yes';
```
Maybe rename it to something like `OPTION_VALUE_YES` to make it more indicative that it's a constant of a state value.

ecgan replied:
Done in [`162d790` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/162d790a7afb4385eab63d8407ef70c1e2e530f1).

eason9487 commented on plugins/woocommerce-admin/client/marketing/hooks/useIntroductionBanner.ts:
```diff
@@ -0,0 +1,40 @@
+	return useSelect( ( select ) => {
+		const { getOption, isOptionsUpdating, hasFinishedResolution } =
+			select( OPTIONS_STORE_NAME );
+		const isUpdateRequesting = isOptionsUpdating();
```
The state returned by `isOptionsUpdating()` is a global state of the `OPTIONS_STORE_NAME` store. Therefore, any updates to the store will change this state.

When other places on the same page also use this store to update WP Options, the usage here leads to the following situation.

https://user-images.githubusercontent.com/17420811/223936181-347300d9-36aa-417c-ac99-f96214bfa080.mp4

<br />

Or, it can be simulated by running the following code via the Console tab of DevTool:

```js
wp.data.dispatch( 'wc/admin/options' ).updateOptions( { __dev_test: 'yes' } )
```

ecgan replied:
Thanks for catching that, I didn't realize that. I actually got it from the existing code in the classic marketing page: 😅 

https://github.com/woocommerce/woocommerce/blob/c5564a15c11eae5b30de7ab6e37edbd973a29ebe/plugins/woocommerce-admin/client/marketing/overview/welcome-card/index.js#L70-L77

I have fixed it in [`99c6ebf` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/99c6ebff1554b8ed7fd111af20395e9b8c7cb0c1). I found that we can simply rely on `getOption`, because when we call `updateOptions`, the options will be updated in the data store immediately before API is called to save the value in the database:

https://github.com/woocommerce/woocommerce/blob/c5564a15c11eae5b30de7ab6e37edbd973a29ebe/packages/js/data/src/options/actions.ts#L42-L51

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.scss:
```diff
@@ -0,0 +1,50 @@
+
+		.woocommerce-marketing-introduction-banner-close-button {
+			position: absolute;
+			top: $gap-smaller;
```
This spacing looks closer to `$gap-small` ?

ecgan replied:
Thanks, sharp eyes! Done in [`65523e3` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/65523e36dda2f0758e5917e0e275b744a36ea3b1).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.tsx:
```diff
@@ -0,0 +1,165 @@
+						<Flex>
+							<Icon
+								icon={
+									<img
```
This `<Icon>` could be omitted.

ecgan replied:
Done in [`155211d` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/155211d77ac41caf3143a77f483ee1f43bc227ce).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.tsx:
```diff
@@ -0,0 +1,165 @@
+	onDismiss,
+	onAddChannels,
+}: IntroductionBannerProps ) => {
+	const [ open, setOpen ] = useState( false );
```
Perhaps replacing `open` with a name that more clearly conveys the state.

ecgan replied:
Done in [`6b75b5d` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/6b75b5db384de2a9c9d9d5c3f2b37a3bc55a5617).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.tsx:
```diff
@@ -0,0 +1,165 @@
+				</Button>
+				<img
+					src={
+						showCreateCampaignButton || showAddChannelsButton
+							? illustrationLargeUrl
+							: illustrationUrl
```
It should be possible to place a background image via CSS to eliminate an almost duplicate image.

ecgan replied:
Yeah I thought about that, but the problem is with the gradient background in the SVG files. I think it would be better if we have just the product image in the SVG, and have the gradient background applied onto HTML div via CSS. I'll leave it as a future enhancement PR.

eason9487 replied:
> I think it would be better if we have just the product image in the SVG, and have the gradient background applied onto HTML div via CSS.

This could be done by these steps:

1. Remove the gradient background from the illustration-large.svg by deleting line 3: `<rect width="271.419" height="260" ...` and lines 65-68: `<linearGradient ... </linearGradient>`.
   - The illustration-large.svg file after removing the gradient background:
      ![illustration-large](https://user-images.githubusercontent.com/17420811/226282498-dc20e873-a373-4865-879a-0f7baa57ff1f.svg)
2. Add the gradient background removed from SVG to CSS:
   ```scss
   .woocommerce-marketing-introduction-banner-illustration {
   	flex: 0 0 270px;
   	background: linear-gradient(90deg, rgba(247, 237, 247, 0) 5.31%, rgba(196, 152, 217, 0.12) 77.75%),
   		linear-gradient(90deg, rgba(247, 237, 247, 0) 22%, rgba(196, 152, 217, 0.12) 84.6%);
   
   	.woocommerce-marketing-introduction-banner-image-placeholder {
   		width: 100%;
   		height: 100%;
   		background: center / cover no-repeat;
   	}
   ```
   - or this version if want to make it pretty close to the current visual result:
      ```scss
      .woocommerce-marketing-introduction-banner-image-placeholder {
      	width: 100%;
      	height: 100%;
      	background: center / contain no-repeat;
      
      	&--smaller {
      		background-size: cover;
      		background-position-y: -17px;
      	}
      }
      ```
3. Replace the [`<img ... />`](https://github.com/woocommerce/woocommerce/blob/5455abcabb31ce7ecf4261711b22cdd4b3959bc9/plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.tsx#L144-L154) with :
   ```js
   <div
   	className="woocommerce-marketing-introduction-banner-image-placeholder"
   	style={ {
   		backgroundImage: `url("${ illustrationLargeUrl }")`,
   	} }
   />
   ```

This gets almost the same results: (The left column is the original, and the right is the replacement)

![2023-03-20 15 52 03](https://user-images.githubusercontent.com/17420811/226282395-e1e7a762-37d9-42dc-a520-f27b973eee42.png)

---

> I'll leave it as a future enhancement PR.

As these SVG files were newly added by this PR, I would suggest adjusting them in this PR. After all, the amount of adjustment needed is not significant, and the effort spent on opening a follow-up issue and then creating another PR to improve it would likely be more.

ecgan replied:
>> I think it would be better if we have just the product image in the SVG, and have the gradient background applied onto HTML div via CSS.
>
> This could be done by these steps: [...]

Thanks @eason9487 for the detailed steps! You could have committed and contributed the code. 😂 I made some minor changes (done in https://github.com/woocommerce/woocommerce/pull/37110/commits/4e38969ce7d2f3b1246573fac46c4eb7b4d3b9af) and this is how it looks, which I think is acceptable and not much difference with the design:

![image](https://user-images.githubusercontent.com/417342/226312643-2b758ed5-8338-4918-9919-7b41ffd3dcd0.png)

I chose to use `background: center / contain no-repeat;` because I don't want the image to be stretched by `background: center / cover no-repeat;`, and I didn't use the `&--smaller` rule because I think that is not needed.

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.tsx:
```diff
@@ -0,0 +1,165 @@
+
+export const IntroductionBanner = ( {
+	onDismiss,
+	onAddChannels,
```
Suggest renaming `onDismiss` to `onRequestDismiss` or `onDismissClick` as this component doesn't handle the actual dismiss itself. The same goes for the `onAddChannels` prop.

ecgan replied:
Done in [`c6a95ae` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/c6a95ae9bf47ea38a6c9a1fa654a8fd3ee561f0a).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx:
```diff
@@ -44,11 +54,22 @@ export const MarketingOverviewMultichannel: React.FC = () => {
 
 	return (
 		<div className="woocommerce-marketing-overview-multichannel">
+			{ ! isIntroductionBannerDismissed && (
+				<IntroductionBanner
```
As per the design document, when there are no campaigns:
- if the introduction banner is shown, the campaigns card should be hidden;
- if the introduction banner is hidden, the campaigns card should be shown.

Is this missed or will it be handled in a subsequent PR?

![image](https://user-images.githubusercontent.com/17420811/223959372-35db6167-7dba-4fbe-9cd3-fe13e5a280ea.png)

ecgan replied:
Thanks, it was missed, I didn't notice the condition to display / hide it. Fixed in [`ddd287c` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/ddd287cc4cfb5c2a8de186c7d6b16f86860da292).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.scss:
```diff
@@ -0,0 +1,50 @@
+
+		.woocommerce-marketing-introduction-banner-title {
+			font-size: 20px;
+			font-weight: 400;
```
Looks like this could be omitted.

ecgan replied:
Done in [`38370cc` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/38370cc5449a6debd54cc844e97bc8dc19ead3b1).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.scss:
```diff
@@ -0,0 +1,50 @@
+
+		.woocommerce-marketing-introduction-banner-features {
+			font-size: 13px;
+			font-weight: 400;
```
Looks like these could be omitted.

ecgan replied:
Done in [`38370cc` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/38370cc5449a6debd54cc844e97bc8dc19ead3b1).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/IntroductionBanner/IntroductionBanner.scss:
```diff
@@ -0,0 +1,50 @@
+	}
+
+	.woocommerce-marketing-introduction-banner-illustration {
+		width: 271px;
```
How about setting it with `flex: 0 1 270px;` and `flex: 0 0;` for .woocommerce-marketing-introduction-banner-content instead of two hard-coded values?

ecgan replied:
I tried your suggestion but it didn't work out quite well, there is an empty space on the right:

![image](https://user-images.githubusercontent.com/417342/226194914-6e104929-fb42-4c37-aa1a-00d8ad449bd4.png)

I did a little bit more modification in [`75c11a6` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/75c11a681db3fe7ee13f1446a4462531db6f8a89), so now we have only one hardcoded `270px` value, without the other hardcoded `350px` value:

![image](https://user-images.githubusercontent.com/417342/226194955-d999198c-c3f1-4628-91b0-b360c62c945d.png)

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/Channels/Channels.tsx:
```diff
@@ -81,6 +83,7 @@ export const Channels: React.FC< ChannelsProps > = ( {
 							<CardDivider />
 							<CardBody>
 								<Button
+									ref={ addChannelsButtonRef }
```
For non-leaf components like `Channels`, instead of exposing the real DOM to the outside of this component, it would be a good practice to expose a custom handle of a component via the `useImperativeHandle` hook. So that the dependency between components is on the interface of a component rather than on the DOM instance inside a component, and `MarketingOverviewMultichannel` would no need to know these statements:

https://github.com/woocommerce/woocommerce/blob/6d72c2bbe13573433f96dfa1df09e0c839e0fc03/plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx#L61-L63

Another benefit is that this also makes it possible to replace the direct call to `button.click()` with updating `expanded` state. In the future, it would be easier to add event tracking such as toggle tracking if needed.

ecgan replied:
Thanks so much, this is the first time I learn and use `useImperativeHandle`! Done in [`5455abc` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/5455abcabb31ce7ecf4261711b22cdd4b3959bc9).

eason9487 reviewed (commented):
Thanks for the work.

The banner works generally correct. There are three main concerns:
1. The use of `isOptionsUpdating`
2. The case when there are no campaigns seems to be missed.
3. A illustration image could be eliminated.

The rest are minor things.

ecgan committed:
Merge branch 'feature/34909-marketing-create-campaign-modal' into feature/34904-marketing-introduction-banner

Conflicts:
	plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/Campaigns.test.tsx
	plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/Campaigns.tsx
	plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx

ecgan committed:
Merge branch 'trunk' into feature/34904-marketing-introduction-banner

Conflicts:
	plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx

ecgan committed:
Merge branch 'trunk' into feature/34904-marketing-introduction-banner

ecgan committed:
Specify dependency for useSelect in useIntroductionBanner.

Co-authored-by: Eason <eason.su.tw@gmail.com>

ecgan committed:
Update option name and value to make it more indicative.

ecgan committed:
Remove the use of isOptionsUpdating in useIntroductionBanner.

isOptionsUpdating will return true for any option update, not just our option here. This causes issue as shown in demo video in https://github.com/woocommerce/woocommerce/pull/37110#discussion_r1130531104.

We can just depend on getOption. When we update an option, it will be updated immediately in wp.data store before making API request to update the option in database (see https://github.com/woocommerce/woocommerce/blob/c5564a15c11eae5b30de7ab6e37edbd973a29ebe/packages/js/data/src/options/actions.ts#L44).

ecgan committed:
Fix useSelect dependency in useIntroductionBanner.

ecgan committed:
Fix spacing for close button in IntroductionBanner.

ecgan committed:
Remove unneeded Icon component.

ecgan committed:
Rename variables to make them clearer.

ecgan committed:
Rename callback props in IntroductionBanner to make them clearer.

ecgan committed:
Move useCampaigns to shared hooks directory.

ecgan committed:
Set default parameter values for useCampaigns.

ecgan committed:
Show Campaigns card when banner is dismissed or campaigns total is truthy.

ecgan committed:
Remove unneeded CSS.

ecgan committed:
Simplify CSS to have less hardcoded width value.

ecgan committed:
Use useImperativeHandle instead of exposing button ref in Channels.

ecgan commented:
@eason9487 , thanks so much for the review! I have addressed the code review comments. This is now ready for another round of review. 🙂  🙏

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx:
```diff
@@ -36,15 +46,22 @@ export const MarketingOverviewMultichannel: React.FC = () => {
 	}
 
+	const shouldShowCampaigns =
+		dataRegistered?.length &&
```
Since the naming of `shouldShowCampaigns` appears to be a boolean value, suggest making it definitely a boolean value so that the below use would not need the type casting and the semantics would be closer to the actual implementation results.

https://github.com/woocommerce/woocommerce/blob/5455abcabb31ce7ecf4261711b22cdd4b3959bc9/plugins/woocommerce-admin/client/marketing/overview-multichannel/MarketingOverviewMultichannel.tsx#L84

ecgan replied:
Done in [`2929990` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/29299903c3f31ca3307ba63e172b07089c77facc).

eason9487 commented on plugins/woocommerce-admin/client/marketing/overview-multichannel/Channels/Channels.tsx:
```diff
@@ -32,72 +38,95 @@ type ChannelsProps = {
+				scrollIntoAddChannels: () => {
+					setExpanded( true );
+					addChannelsButtonRef.current?.focus();
+					addChannelsButtonRef.current?.scrollIntoView();
```
With a smaller viewport height, the channels section could be covered by the floating header after calling `scrollIntoView`.

https://user-images.githubusercontent.com/17420811/226249139-4fa7d220-bf1b-47e8-b003-7e1bd651709b.mp4

<br />

Maybe set a proper `scroll-margin-top` to the "Add channels" button, or call `scrollIntoView` with options like:

```js
import { useReducedMotion } from '@wordpress/compose';

// ... omitted

const isReducedMotion = useReducedMotion();

// ... omitted

addChannelsButtonRef.current?.scrollIntoView( {
	behavior: isReducedMotion ? 'auto' : 'smooth',
	block: 'center',
} );
```

ecgan replied:
Done in [`7dd7994` (#37110)](https://github.com/woocommerce/woocommerce/pull/37110/commits/7dd7994e01dacb99bd291d0ab288ea769fb27f64). I actually looked into [`scrollIntoViewOptions`](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) before and I noticed it is experimental, so I didn't use it, but I guess it's okay to use it now to fix this scrolling issue.

eason9487 reviewed (dismissed):
Thanks for the update. The test results are correct and as expected.

However, I would still suggest addressing the almost duplicate images that can be eliminated in this PR. because [it doesn't need much time to fix](https://github.com/woocommerce/woocommerce/pull/37110#discussion_r1141772524). Otherwise, it might take more effort to merge a known problem, open a follow-up, and then re-arrange improvement later. But it should be fine to proceed with the current implementation if there is not much time left for this project.

The remaining comments are minor suggestions and won't block this PR.

ecgan committed:
Make shouldShowCampaigns boolean type.

ecgan committed:
Make scrollIntoView center on the "Add channels" button.

ecgan committed:
Use one illustration instead of two in IntroductionBanner.

ecgan committed:
Remove unused illustration.svg file.

ecgan committed:
Rename illustration-large.svg to illustration.svg.

ecgan commented:
@eason9487 , thanks, I took your suggestions and made the changes, and your review approval has been dismissed. Could you do a review and approve again please? 🙏

eason9487 reviewed (approved):
Thanks for the all work. Testing well, and LGTM.
